upgrade/upgrade-scripts/upgrade

```diff
@@ -178,10 +190,41 @@ function upgrade_not_in_place() {
 	"$IMAGE_PATH/upgrade-container" convert-to-rootfs "$CONTAINER" ||
 		die "failed to convert-to-rootfs '$CONTAINER'"
 
+	#
+	# In order for the "rollback" back command to work, we need to
+	# know which rootfs container to rollback to. Thus, we embed
+	# this information into the rootfs container using a ZFS
+	# property; rollback will read this information, to determine
+	# which rootfs container to rollback to.
+	#
+	zfs set \
+		"$ROLLBACK_PROPERTY=$(get_mounted_rootfs_container)" \
```

Comment:
We should probably call `$(get_mounted_rootfs_container)` on a separate line to prevent bash from swallowing errors.

Reply:
give my latest revision and look, and let me know what you think.